### ISSUE TITLE
webSpoon deployed to AWS Elastic Beanstalk does not run properly

### ISSUE THREAD

**Issue by HiromuHota** [HiromuHota/pentaho-kettle#13]
webSpoon of ver. 0.0.0.3 was successfully deployed to AWS Elastic Beanstalk, but it gave me the following error when I accessed via a web browser.
```
HTTP Status 500 - Could not create entrypoint instance: org.pentaho.di.ui.spoon.Spoon

type Exception report

message Could not create entrypoint instance: org.pentaho.di.ui.spoon.Spoon

description The server encountered an internal error that prevented it from fulfilling this request.

exception

java.lang.RuntimeException: Could not create entrypoint instance: org.pentaho.di.ui.spoon.Spoon
	org.eclipse.rap.rwt.internal.lifecycle.DefaultEntryPointFactory.create(DefaultEntryPointFactory.java:52)
	org.eclipse.rap.rwt.internal.lifecycle.RWTLifeCycle.createUI(RWTLifeCycle.java:176)
	org.eclipse.rap.rwt.internal.lifecycle.RWTLifeCycle$UIThreadController.run(RWTLifeCycle.java:290)
	java.lang.Thread.run(Thread.java:745)
	org.eclipse.rap.rwt.internal.lifecycle.UIThread.run(UIThread.java:107)
root cause

java.lang.IllegalArgumentException: Argument cannot be null
	org.eclipse.swt.SWT.error(SWT.java:3706)
	org.eclipse.swt.SWT.error(SWT.java:3643)
	org.eclipse.swt.SWT.error(SWT.java:3614)
	org.eclipse.swt.widgets.Dialog.checkParent(Dialog.java:292)
	org.eclipse.swt.widgets.Dialog.<init>(Dialog.java:115)
	org.pentaho.di.ui.core.dialog.ErrorDialog.<init>(ErrorDialog.java:97)
	org.pentaho.di.ui.spoon.Spoon.<init>(Spoon.java:735)
	org.pentaho.di.ui.spoon.Spoon.<init>(Spoon.java:714)
	sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	org.eclipse.rap.rwt.internal.util.ClassUtil.createInstance(ClassUtil.java:57)
	org.eclipse.rap.rwt.internal.util.ClassUtil.newInstance(ClassUtil.java:38)
	org.eclipse.rap.rwt.internal.util.ClassUtil.newInstance(ClassUtil.java:32)
	org.eclipse.rap.rwt.internal.lifecycle.DefaultEntryPointFactory.create(DefaultEntryPointFactory.java:49)
	org.eclipse.rap.rwt.internal.lifecycle.RWTLifeCycle.createUI(RWTLifeCycle.java:176)
	org.eclipse.rap.rwt.internal.lifecycle.RWTLifeCycle$UIThreadController.run(RWTLifeCycle.java:290)
	java.lang.Thread.run(Thread.java:745)
	org.eclipse.rap.rwt.internal.lifecycle.UIThread.run(UIThread.java:107)
note The full stack trace of the root cause is available in the Apache Tomcat/8.0.38 logs.

Apache Tomcat/8.0.38
```

**Comment by HiromuHota** [comment]
The Tomcat log says
```
Unable to create default kettle.properties file : /usr/share/tomcat8/.kettle/kettle.properties
[Ljava.lang.StackTraceElement;@1bc20ed4
org.pentaho.di.core.exception.KettleException: 
Unable to read file '/usr/share/tomcat8/.kettle/kettle.properties'
/usr/share/tomcat8/.kettle/kettle.properties (No such file or directory)

        at org.pentaho.di.core.util.EnvUtil.readPropertiesByFullPath(EnvUtil.java:68)
        at org.pentaho.di.core.util.EnvUtil.readProperties(EnvUtil.java:56)
        at org.pentaho.di.core.util.EnvUtil.environmentInit(EnvUtil.java:96)
        at org.pentaho.di.core.KettleClientEnvironment.init(KettleClientEnvironment.java:80)
        at org.pentaho.di.core.KettleEnvironment.init(KettleEnvironment.java:97)
        at org.pentaho.di.ui.spoon.BasicApplication$1.call(BasicApplication.java:61)
        at org.pentaho.di.ui.spoon.BasicApplication$1.call(BasicApplication.java:1)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.FileNotFoundException: /usr/share/tomcat8/.kettle/kettle.properties (No such file or directory)
        at java.io.FileInputStream.open0(Native Method)
        at java.io.FileInputStream.open(FileInputStream.java:195)
        at java.io.FileInputStream.<init>(FileInputStream.java:138)
        at java.io.FileInputStream.<init>(FileInputStream.java:93)
        at org.pentaho.di.core.util.EnvUtil.readPropertiesByFullPath(EnvUtil.java:65)
        ... 10 more
```

**Comment by HiromuHota** [comment]
I ssh logged into the EC2 instance that was used for the Elastic Beanstalk environment, created `.kettle` folder, set permissions properly, and restarted the app server (Tomcat).
The file access permission issue was resolved, but still getting the HTTP 500 error.
It turned out to be two separate issues.

**Comment by HiromuHota** [comment]
I resolved the HTTP 500 error by commenting-out [ErrorDialog](https://github.com/HiromuHota/pentaho-kettle/blob/webspoon/0.0.0.3/ui/src/org/pentaho/di/ui/spoon/Spoon.java#L735):
```
    // Load at least one local Pentaho metastore and add it to the delegating metastore
    //
    metaStore = new DelegatingMetaStore();
    try {
      IMetaStore localMetaStore = MetaStoreConst.openLocalPentahoMetaStore();
      metaStore.addMetaStore( localMetaStore );
      metaStore.setActiveMetaStoreName( localMetaStore.getName() );
      if ( rep != null ) {
        metaStore.addMetaStore( 0, rep.getMetaStore() );
        metaStore.setActiveMetaStoreName( rep.getMetaStore().getName() );
      }
    } catch ( MetaStoreException e ) {
      //new ErrorDialog( shell, "Error opening Pentaho Metastore", "Unable to open local Pentaho Metastore", e );
    }
```

Note that the original issue: a local Pentaho metastore cannot be loaded remains un-resolved.

**Comment by HiromuHota** [comment]
Oh I didn't have to comment-out the ErrorDialog.
I just had to properly set environment variables as follows:
```
KETTLE_HOME=/usr/share/tomcat8/webapps/ROOT/.kettle
PENTAHO_METASTORE_FOLDER=/usr/share/tomcat8/webapps/ROOT/.pentaho
```

The base path `/usr/share/tomcat8/` depends on the app server.
In addition to the above two, if you set 
```
KETTLE_PLUGIN_BASE_FOLDERS=/usr/share/tomcat8/webapps/ROOT/plugins
```
, pdi-pur-plugin can be loaded. This resolved the issue #14 too.
